baremetal/metal3data_manager.go

```diff
@@ -1419,7 +1425,7 @@ func fetchM3IPClaim(ctx context.Context, cl client.Client, mLog logr.Logger,
 	}
 	if err := cl.Get(ctx, metal3ClaimName, metal3IPClaim); err != nil {
 		if apierrors.IsNotFound(err) {
-			errMessage := "Address claim not found, requeuing"
+			errMessage := "Address claim not found"
```

Comment:
We are not deciding at this point if we are requeuing or not. In fact, we are calling this function several times in `ensureM3IPClaim` and not necessarily requeuing at all.